Update of the section on local destination environment (#786)
SAP/cloud-sdk: docs/java/features/connectivity/destinations.mdx

Try<Destination> destination = DestinationAccessor.getLoader().tryGetDestination("MyDestination", options);
```

## Configure HTTP Destinations for Local Development Environment

When testing your app on you local development machine instead of SAP BTP, you require some configuration steps to make the destination retrieval work.

:::note Relevant only for HTTP destinations

Check our [BAPI and RFC documentation](../bapi-and-rfc/overview.mdx) if you want to configure an RFC destination for local development.

:::

To better understand the configuration steps, we will shortly outline how destination retrieval works when your app runs on SAP BTP.

### Background: Destination Retrieval on SAP Business Technology Platform

By default, the SAP Cloud SDK uses a set of destination loaders in the following order when searching for a destination via the `DestinationAccessor` API

1. Environment variable
1. SAP BTP Destination service

Example API call looks like this:

```java

DestinationAccessor.tryGetDestination("MyDestination");

```

<details>

>     <summary>Details and example values of a destination retrieval steps</summary>

Reviewer: ```suggestion
    <summary>Details and example values of a destination retrieval</summary>
```

Try<Destination> destination = DestinationAccessor.getLoader().tryGetDestination("MyDestination", options);
```

## Configure HTTP Destinations for Local Development Environment

When testing your app on you local development machine instead of SAP BTP, you require some configuration steps to make the destination retrieval work.

:::note Relevant only for HTTP destinations

Check our [BAPI and RFC documentation](../bapi-and-rfc/overview.mdx) if you want to configure an RFC destination for local development.

:::

To better understand the configuration steps, we will shortly outline how destination retrieval works when your app runs on SAP BTP.

### Background: Destination Retrieval on SAP Business Technology Platform

By default, the SAP Cloud SDK uses a set of destination loaders in the following order when searching for a destination via the `DestinationAccessor` API

1. Environment variable
1. SAP BTP Destination service

> 1. SAP BTP Connectivity service

Reviewer: The connectivity service is not used as a source for destinations. It may come into play only if the received destination is of type `OnPremise` and, therefore, further proxy authorization is needed.

```suggestion
```

]
```

You can add as many destinations as you need this way.

> :::tip (Beta) Authorization Token Forwarding

Author: @Johannes-Schneider , do you think this is still Beta by the way?

Reviewer: This feature has been added in Sept. 2021 and has not been changed ever since... So IMO it's fair to assume that this is working as expected without the need for (drastic) changes in the foreseeable future.

> :::tip (Beta) Authorization Token Forwarding

Reviewer: ```suggestion
:::tip Authorization Token Forwarding
```

By adding `"forwardAuthToken": "true"` to your destination, you will forward the value of the `Authorization` header from the current request directly to the target system.
:::

1. Lookup via SAP BTP Destination service

>    If the previous lookup was not successful, the next destination loaders examine the environment variable `VCAP_SERVICES` to see if the SAP BTP Destination service is bound.

Reviewer: ```suggestion
   If the previous lookup was not successful, the next destination loader examines the environment variable `VCAP_SERVICES` to see if the SAP BTP Destination service is bound.
```

```bash

npm i -g @sap/cds-dk

```

Now, you can start any Java app with service binding injected into your local environment by using `cds bind` command.
The command will create a `VCAP_SERVICES` variable containing service keys from a Cloud Foundry space or Kubernetes cluster without storing credentials on a disk.
Start your Java application similarly to the code snippet below:

```bash

cds bind --exec mvn spring-boot:run

```

For more details look at the [official CAP documentation](https://cap.cloud.sap/docs/advanced/hybrid-testing#run-with-service-bindings)

### Configure Destinations Manually

#### Destinations With Proxy Type `Internet`

Create the destination configuration and pass it via the environment variable `destinations`.
Check the example below:

```json
[
  {
    "type": "HTTP",
    "name": "MyDestination",
    "proxyType": "Internet",
    "description": "This destination rocks!",
    "authentication": "BasicAuthentication",
    "url": "https://URL",
    "user": "USER",
    "password": "PASSWORD"
  }
]
```

You can add as many destinations as you need by adding them to the `JSON` array.

> :::tip (Beta) Authorization Token Forwarding

Reviewer: ```suggestion
:::tip Authorization Token Forwarding
```

]
```

You can add as many destinations as you need this way.

:::tip (Beta) Authorization Token Forwarding
By adding `"forwardAuthToken": "true"` to your destination, you will forward the value of the `Authorization` header from the current request directly to the target system.
:::

1. Lookup via SAP BTP Destination service

   If the previous lookup was not successful, the next destination loaders examine the environment variable `VCAP_SERVICES` to see if the SAP BTP Destination service is bound.

>    The SAP Cloud SDK tries to fetch the destination from the SAP BTP Destination service the destination name.

Reviewer: ```suggestion
   The SAP Cloud SDK tries to fetch the destination from the SAP BTP Destination service using the given destination name.
```

> 1. Lookup via SAP BTP Connectivity Service

Reviewer: As explained above, this is not really an "alternative" step with regards to getting a destination.
It is rather an **additional** step that is required in case we need to reach the target system via the connectivity service (a.k.a. the destination is of type `OnPremise`).
So I don't think the numeration makes much sense here. 

Maybe something like the following is more suitable?

```suggestion
**_Reaching On Premise Systems via the BTP Connectivity Service_**
```

   In case the destination is of proxy type `OnPremise`, the SAP Cloud SDK performs an additional query to the SAP BTP Connectivity service to obtain further proxy information and an authorization token.

</details>

### Configuration Steps for Local Development Environment

> You can either try CAP Development Kit for automated setup or configure everything manually.

Reviewer: Does this chapter really provide any value?
I'd suggest to simply remove it.

### Configure Destinations Manually

#### Destinations With Proxy Type `Internet`

Create the destination configuration and pass it via the environment variable `destinations`.
Check the example below:

```json
[
  {
    "type": "HTTP",
    "name": "MyDestination",
    "proxyType": "Internet",
    "description": "This destination rocks!",
    "authentication": "BasicAuthentication",
    "url": "https://URL",
    "user": "USER",
    "password": "PASSWORD"
  }
]
```

You can add as many destinations as you need by adding them to the `JSON` array.

:::tip (Beta) Authorization Token Forwarding

By adding `"forwardAuthToken": "true"` to your destination, you will forward the value of the `Authorization` header from the current request directly to the target system.

:::

#### Destinations With Proxy Type `OnPremise`

For this destination types we need to access the **Destination** and **Connectivity** services on SAP BTP.
To achieve that you have to bind them and inject via [CAP Development Kit](#automatically-inject-service-bindings-via-cap-development-kit) or by manually providing your [`VCAP_SERVIES`](#create-and-provide-vcap_services-variable) variable.

> :::tip when your client and On-premise service are in the same network

Reviewer: Title case

```suggestion
:::tip When your Client and On-Premise Service are in the Same Network
```

Author: The admonition headers are capitalized by default:))) I had to capitalize OP because of our rather strict and not always logical vale rules.

> If your local development environment and an On-premise service are in the same network, you can use a simple HTTP destination by providing it via `destinations` environment variable instead of relying on the SAP BTP Connectivity service.

Reviewer: ```suggestion
If your local development environment and an On-Premise service are in the same network, you can use a simple HTTP destination by providing it via `destinations` environment variable instead of relying on the SAP BTP Connectivity service.
```
